docs/developer/meta/ci/testing.md

```diff
@@ -0,0 +1,191 @@
+# Testing
+
+-----
+
+## Workflows
+
+- [Master](https://github.com/DataDog/integrations-core/blob/master/.github/workflows/master.yml) - Runs tests on Python 3 for [every target](#target-enumeration) on merges to the `master` branch
+- [PR](https://github.com/DataDog/integrations-core/blob/master/.github/workflows/pr.yml) - Runs tests on Python 2 & 3 for any modified target in a pull request as long as the base or developer packages were not modified
+- [PR All](https://github.com/DataDog/integrations-core/blob/master/.github/workflows/pr-all.yml) - Runs tests on Python 2 & 3 for [every target](#target-enumeration) in a pull request if the base or developer packages were modified
+- [Nightly minimum base package test](https://github.com/DataDog/integrations-core/blob/master/.github/workflows/nightly-base-package.yml) - Runs tests for [every target](#target-enumeration) once nightly using the minimum declared required version of the base package
+- [Nightly Python 2 tests](https://github.com/DataDog/integrations-core/blob/master/.github/workflows/nightly-py2.yml) - Runs tests on Python 2 for [every target](#target-enumeration) once nightly
+- [Test Agent release](https://github.com/DataDog/integrations-core/blob/master/.github/workflows/test-agent.yml) - Runs tests for [every target](#target-enumeration) when manually scheduled using specific versions of the Agent for E2E tests
+
+## Reusable workflows
+
+These can be [used](https://docs.github.com/en/actions/using-workflows/reusing-workflows) by other repositories.
+
+### PR test
+
+This [workflow](https://github.com/DataDog/integrations-core/blob/master/.github/workflows/pr-test.yml) is meant to be used on pull requests.
+
+First it [computes the job matrix](https://github.com/DataDog/integrations-core/blob/master/.github/workflows/compute-matrix.yml) based on what was changed. Since this is time sensitive, rather than fetching the entire history we use GitHub's API to find out the precise depth to fetch in order to reach the merge base. Then it runs the [test workflow](#test-target) for every job in the matrix.
+
+!!! note
+    Changes that match any of the following patterns inside a directory will trigger the testing of that target:
+
+    - `assets/configuration/**/*`
+    - `tests/**/*`
+    - `*.py`
+    - `hatch.toml`
+    - `pyproject.toml`
```

Comment:
Not related to this doc but I think we should also run the tests when we modify the `metadata.csv` because our tests can validate it with `assert_metrics_using_metadata`. What do you think?

Reply:
Good idea! https://github.com/DataDog/integrations-core/pull/14429